api/src/test/java/org/pmiops/workbench/api/CohortsControllerTest.java

```diff
@@ -440,61 +479,185 @@ public void testCreateCohortBadCriteria() {
+    assertForbiddenException(exception);
+  }
 
-    Cohort newCohort =
-        cohortsController
-            .duplicateCohort(workspace.getNamespace(), workspace.getId(), params)
-            .getBody();
-    newCohort =
-        cohortsController
-            .getCohort(workspace.getNamespace(), workspace.getId(), newCohort.getId())
-            .getBody();
+  @Test
+  public void testUpdateCohortAccessLevelReader() {
+    Cohort saved = createTestCohort(WorkspaceAccessLevel.WRITER);
+    // update
+    stubGetWorkspace(
+        workspace.getNamespace(), workspace.getName(), CREATOR_EMAIL, WorkspaceAccessLevel.READER);
+    stubGetWorkspaceAcl(
+        workspace.getNamespace(), workspace.getName(), CREATOR_EMAIL, WorkspaceAccessLevel.READER);
+    Cohort toUpdate = saved.name(UPDATED_COHORT_NAME);
+    CLOCK.increment(1000L); // lets say time ticked 1 sec past
+    Throwable exception =
+        assertThrows(
+            ForbiddenException.class,
+            () ->
+                cohortsController.updateCohort(
+                    workspace.getNamespace(), workspace.getId(), toUpdate.getId(), toUpdate));
+    assertForbiddenException(exception);
+  }
+
+  @Test
+  public void testUpdateCohortAccessLevelNoAccess() {
+    Cohort saved = createTestCohort(WorkspaceAccessLevel.WRITER);
+    // update
+    stubGetWorkspace(
+        workspace.getNamespace(),
+        workspace.getName(),
+        CREATOR_EMAIL,
+        WorkspaceAccessLevel.NO_ACCESS);
+    stubGetWorkspaceAcl(
+        workspace.getNamespace(),
+        workspace.getName(),
+        CREATOR_EMAIL,
+        WorkspaceAccessLevel.NO_ACCESS);
+    Cohort toUpdate = saved.name(UPDATED_COHORT_NAME);
+    CLOCK.increment(1000L); // lets say time ticked 1 sec past
+    Throwable exception =
+        assertThrows(
+            ForbiddenException.class,
+            () ->
+                cohortsController.updateCohort(
+                    workspace.getNamespace(), workspace.getId(), toUpdate.getId(), toUpdate));
+    assertForbiddenException(exception);
+  }
+
+  private void assertForbiddenException(Throwable exception) {
+    assertThat(exception)
+        .hasMessageThat()
+        .isEqualTo(
+            String.format(
+                "You do not have sufficient permissions to access workspace %s/workspace",
+                workspace.getNamespace()));
+  }
 
-    assertThat(newCohort.getName()).isEqualTo(params.getNewName());
-    assertThat(newCohort.getCriteria()).isEqualTo(originalCohort.getCriteria());
-    assertThat(newCohort.getType()).isEqualTo(originalCohort.getType());
-    assertThat(newCohort.getDescription()).isEqualTo(originalCohort.getDescription());
+  @Test
+  public void testDuplicateCohortAccessLevelOwner() {
+    Cohort original = createTestCohort(WorkspaceAccessLevel.OWNER);
+    CLOCK.increment(1000L); // increment clock before duplicating
+    Cohort duplicated = createDuplicateTestCohort(WorkspaceAccessLevel.OWNER, original);
```

Comment:
This method suffers the same problem as createTestCohort(). Please split into more logical methods or be descriptive in name about what this method really does.

Reply:
updated code to address these issues